app/src/components/domain/RiskImminentEvents/WfpAdam/index.tsx

```diff
@@ -83,6 +117,48 @@ function WfpAdam(props: Props) {
         apiType: 'risk',
         url: '/api/v1/adam-exposure/{id}/exposure/',
         pathVariables: ({ eventId }) => ({ id: Number(eventId) }),
+        onSuccess: (res) => {
+            if (isNotDefined(res)) {
+                return defaultLayersValue;
+            }
+
+            const { storm_position_geojson: stormPositionGeoJson } = res;
+
+            if (isNotDefined(stormPositionGeoJson)) {
+                return defaultLayersValue;
+            }
+
+            const stormPositions = isValidFeatureCollection(stormPositionGeoJson)
+                ? stormPositionGeoJson
+                : undefined;
+
+            const updatedLayers = { ...defaultLayersValue };
+
+            stormPositions?.features?.forEach((feature) => {
+                const { geometry, properties } = feature;
+
+                if (geometry.type === 'Point' || geometry.type === 'MultiPoint') {
+                    updatedLayers[LAYER_CYCLONE_NODES] = true;
+                }
+
+                if (geometry.type === 'LineString' || geometry.type === 'MultiLineString') {
+                    updatedLayers[LAYER_CYCLONE_TRACKS] = true;
+                }
+                if (geometry.type === 'Polygon' || geometry.type === 'MultiPolygon') {
+                    updatedLayers[LAYER_CYCLONE_BUFFERS] = true;
+                }
+
+                if (properties?.alert_level === 'Cones') {
+                    updatedLayers[LAYER_CYCLONE_UNCERTAINTY] = true;
+                }
+                return updatedLayers as typeof defaultLayersValue;
```

Comment:
do we need to cast here?

Reply:
no we don't need to cast here, I have removed now.